tests/dataimportcron_test.go

```diff
@@ -28,90 +29,118 @@ const (
-		} else {
-			url = trustedRegistryURL()
-			err = utils.AddInsecureRegistry(f.CrClient, url)
-			Expect(err).To(BeNil())
+		By(fmt.Sprintf("Create labeled DatVolume %s for garbage collection test", dvName))
+		dv := utils.NewDataVolumeWithRegistryImport(dvName, "5Gi", "")
+		dv.Spec.Source.Registry = reg
+		dv.Labels = map[string]string{common.DataImportCronLabel: cronName}
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, ns, dv)
+		Expect(err).ToNot(HaveOccurred())
 
-			hasInsecReg, err := utils.HasInsecureRegistry(f.CrClient, url)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(hasInsecReg).To(BeTrue())
-			defer utils.RemoveInsecureRegistry(f.CrClient, url)
+		By(fmt.Sprintf("Create new DataImportCron %s, url %s", cronName, *reg.URL))
+		cron = NewDataImportCron(cronName, "5Gi", scheduleEveryMinute, dataSourceName, *reg)
+		if !garbageCollection {
+			garbageCollect := cdiv1.DataImportCronGarbageCollectNever
+			cron.Spec.GarbageCollect = &garbageCollect
+		}
+		if !retention {
+			retentionPolicy := cdiv1.DataImportCronRetainNone
+			cron.Spec.RetentionPolicy = &retentionPolicy
 		}
-
-		cron = NewDataImportCron(cronName, "5Gi", schedule, dataSourceName, cdiv1.DataVolumeSourceRegistry{URL: &url, PullMethod: &registryPullNode}, retentionPolicy)
-		By(fmt.Sprintf("Create new DataImportCron %s", url))
 		cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Create(context.TODO(), cron, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
-		if schedule == scheduleEveryMinute {
-			By("Verify cronjob was created")
+		By("Verify cronjob was created")
+		Eventually(func() bool {
+			_, err := f.K8sClient.BatchV1().CronJobs(f.CdiInstallNs).Get(context.TODO(), controller.GetCronJobName(cron), metav1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return false
+			}
+			Expect(err).ToNot(HaveOccurred())
+			return true
+		}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+
+		waitForConditions := func(statusProgressing, statusUpToDate corev1.ConditionStatus) {
+			By(fmt.Sprintf("Wait for DataImportCron Progressing:%s, UpToDate:%s", statusProgressing, statusUpToDate))
 			Eventually(func() bool {
-				_, err = f.K8sClient.BatchV1beta1().CronJobs(f.CdiInstallNs).Get(context.TODO(), controller.GetCronJobName(cron), metav1.GetOptions{})
-				if errors.IsNotFound(err) {
-					return false
-				}
-				return err == nil
+				var err error
+				cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Get(context.TODO(), cronName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				condProgressing := controller.FindDataImportCronConditionByType(cron, cdiv1.DataImportCronProgressing)
+				condUpToDate := controller.FindDataImportCronConditionByType(cron, cdiv1.DataImportCronUpToDate)
+				return condProgressing != nil && condProgressing.Status == statusProgressing &&
+					condUpToDate != nil && condUpToDate.Status == statusUpToDate
 			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
```

Comment:
the same here, a description in Eventually/Should, helps read test reports (especially on failed tests)

Reply:
Added for all Eventually/Should around.